src/Sulu/Bundle/SecurityBundle/User/UserProvider.php

```diff
@@ -103,4 +107,11 @@ public function supportsClass($class)
     {
         return \is_subclass_of($class, UserInterface::class);
     }
+
+    public function upgradePassword(PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void
+    {
+        $user->setPassword($newHashedPassword);
+
+        $this->doctrine->getManager()->flush();
```

Comment:
we should flush over the entityManager service.

```suggestion
        Assert::notNull($this->entityManager, 'Entity manager is required for upgradePassword method.');
        
        $this->entityManager->flush();
```